File: iosApp/iosAppTests/Rewards/RewardsStateRobot.swift
import ComposeApp

final class RewardsStateRobot: BaseStateRobot<RewardsUiState, RewardsEvent> {

    override func defaultState() -> RewardsUiState {
        RewardsUiState(
            progress: RewardsProgress(currentPoints: 2450, nextRewardName: "Free Fries", pointsToNextReward: 550, progressFraction: 0.82),
            vaultSpecials: [
                VaultSpecial(id: "1", title: "Double Quarter", pointsCost: "3000 PTS", imageUrl: "", tag: "LIMITED", isFeatured: true),
                VaultSpecial(id: "2", title: "McFlurry", pointsCost: "1500 PTS", imageUrl: "", tag: nil, isFeatured: false),
            ],
            history: [
                HistoryEntry(id: "1", title: "Big Mac Meal", subtitle: "Yesterday", points: "+150", isPositive: true, icon: "🍔"),
            ],
            eventSink: createEventSink()
        )
    }

    func stateWithNoProgress() -> RewardsUiState {
        RewardsUiState(
            progress: nil,
            vaultSpecials: [
                VaultSpecial(id: "1", title: "Double Quarter", pointsCost: "3000 PTS", imageUrl: "", tag: "LIMITED", isFeatured: true),
            ],
            history: [
                HistoryEntry(id: "1", title: "Big Mac Meal", subtitle: "Yesterday", points: "+150", isPositive: true, icon: "🍔"),
            ],
            eventSink: createEventSink()
        )
    }

    func stateWithEmptyVault() -> RewardsUiState {
        RewardsUiState(
            progress: RewardsProgress(currentPoints: 2450, nextRewardName: "Free Fries", pointsToNextReward: 550, progressFraction: 0.82),
            vaultSpecials: [],
            history: [
                HistoryEntry(id: "1", title: "Big Mac Meal", subtitle: "Yesterday", points: "+150", isPositive: true, icon: "🍔"),
            ],
            eventSink: createEventSink()
        )
    }

    func stateWithEmptyHistory() -> RewardsUiState {
        RewardsUiState(
            progress: RewardsProgress(currentPoints: 2450, nextRewardName: "Free Fries", pointsToNextReward: 550, progressFraction: 0.82),
            vaultSpecials: [
                VaultSpecial(id: "1", title: "Double Quarter", pointsCost: "3000 PTS", imageUrl: "", tag: "LIMITED", isFeatured: true),
            ],
            history: [],
            eventSink: createEventSink()
        )
    }
}

File: features/rewards/impl/presentation/src/androidDeviceTest/kotlin/com/mockdonalds/app/features/rewards/presentation/RewardsStateRobot.kt
package com.mockdonalds.app.features.rewards.presentation

import com.mockdonalds.app.core.test.StateRobot
import com.mockdonalds.app.features.rewards.api.domain.HistoryEntry
import com.mockdonalds.app.features.rewards.api.domain.RewardsProgress
import com.mockdonalds.app.features.rewards.api.domain.VaultSpecial

class RewardsStateRobot : StateRobot<RewardsUiState, RewardsEvent>() {

    override fun defaultState() = RewardsUiState(
        progress = RewardsProgress(
            currentPoints = 1250,
            nextRewardName = "Free Fries",
            pointsToNextReward = 250,
            progressFraction = 0.83f,
        ),
        vaultSpecials = listOf(
            VaultSpecial(id = "1", title = "Secret Burger", pointsCost = "2000 PTS", imageUrl = "", tag = "LIMITED", isFeatured = true),
            VaultSpecial(id = "2", title = "Gold Shake", pointsCost = "1500 PTS", imageUrl = "", tag = null, isFeatured = false),
        ),
        history = listOf(
            HistoryEntry(id = "1", title = "Big Mac Meal", subtitle = "Yesterday", points = "+150", isPositive = true, icon = "🍔"),
        ),
        eventSink = createEventSink(),
    )

    fun stateWithNoProgress() = defaultState().copy(
        progress = null,
        eventSink = createEventSink(),
    )

    fun stateWithEmptyVault() = defaultState().copy(
        vaultSpecials = emptyList(),
        eventSink = createEventSink(),
    )

    fun stateWithEmptyHistory() = defaultState().copy(
        history = emptyList(),
        eventSink = createEventSink(),
    )
}
